manager/api_sdk_manager.go

```diff
@@ -0,0 +1,118 @@
+package manager
+
+import (
+	"encoding/json"
+	"net/http"
+
+	"github.com/companieshouse/api-sdk-go/apikey"
+	sdk "github.com/companieshouse/api-sdk-go/companieshouseapi"
+	choauth2 "github.com/companieshouse/api-sdk-go/oauth2"
+	"github.com/companieshouse/api-sdk-manager-go-library/config"
+	"github.com/pkg/errors"
+	goauth2 "golang.org/x/oauth2"
+)
+
+var basePathOverridden bool = false
+
+// GetSDK will return an instance of the Go SDK using an oauth2 authenticated
+// HTTP client if possible, else an API-key authenticated HTTP client will be used
+func GetSDK(req *http.Request) (*sdk.Service, error) {
+
+	cfg, err := config.Get()
+	if err != nil {
+		return nil, err
+	}
+
+	// Override BasePath here to route API requests via ERIC
+	if !basePathOverridden && len(cfg.APIURL) > 0 {
+		sdk.BasePath = cfg.APIURL
+		basePathOverridden = true
+	}
+
+	token := decodePassthroughHeader(req)
+
+	httpClient, err := getHttpClient(token, req)
```

Comment:
Would it be a good idea calling `decodePassthroughHeader(req)` from within `getHttpClient` so that we abstract it and only need a request as a parameter?